packages/patternfly-4/react-core/src/components/Select/Select.tsx

```diff
@@ -1,89 +1,103 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
 import styles from '@patternfly/react-styles/css/components/Select/select';
 import badgeStyles from '@patternfly/react-styles/css/components/Badge/badge';
 import formStyles from '@patternfly/react-styles/css/components/FormControl/form-control';
 import buttonStyles from '@patternfly/react-styles/css/components/Button/button';
 import { css } from '@patternfly/react-styles';
 import { TimesCircleIcon } from '@patternfly/react-icons';
-import { Chip, ChipGroup } from '../ChipGroup';
-import SingleSelect from './SingleSelect';
-import CheckboxSelect from './CheckboxSelect';
-import SelectToggle from './SelectToggle';
-import SelectOption from './SelectOption';
+import { SelectMenu } from './SelectMenu';
+import { SelectOption } from './SelectOption';
+import { SelectToggle } from './SelectToggle';
 import { SelectContext, SelectVariant } from './selectConstants';
-import { getNextIndex } from '../../helpers/util';
+import { Chip, ChipGroup } from '../ChipGroup';
+import { keyHandler, getNextIndex } from '../../helpers/util';
 
 // seed for the aria-labelledby ID
 let currentId = 0;
 
-const propTypes = {
+export interface SelectProps {
```

Comment:
missing `extends React.HTMLProps<HTMLDivElement>`.

Reply:
Works with Omit for several properties (onSelect, ref, checked, selected). I think it's mostly because I used them differently from what the extension expects, or in the case of ref if can be different types based on which variant is chosen.